types/sinon/ts3.1/index.d.ts

```diff
@@ -1485,17 +1485,17 @@ declare namespace Sinon {
         /**
          * Creates a basic fake, with no behavior
          */
-        (): SinonSpy;
+        (): SinonSpy<any[], void>;
```

Comment:
for backwards compat this should probably be `<any[], any>`

Reply:
Could you give an example of code that would be broken by this change? I don't see it. (Surely if you're just calling `sinon.fake()`, you're not passing it anywhere that expects it to return a value -- because it doesn't.)

Reply:
we don't know where consumers of these types are referencing fakes from. its very possible someone has an explicitly typed variable holding the fake which will differ from this one.

its also possible someone has used the return value somewhere, even though logically it wouldn't do anything/make sense. basically the point here is we don't know. the fact that it is possible means someone somewhere has done it, more than likely, so we have to account for that.

with the previous types, a lot more cases would be valid where you pass a fake into something. i imagine there are people who have been passing fakes into handlers, methods, etc which in fact consume something non-void usually.

your type here is indeed right, and i wish it could be that, but we all know someone somewhere will be depending on the fact it is currently `any`.

Reply:
By the same argument, the type for `sinon.fake.returns` should be `() => SinonSpy<any[], any>`, since somebody may be passing a fake which returns a number into a place which requires a function returning a string (for example).

It is a fair point about maintaining backwards compatibility with the types, and maybe you are right that we should try not to break these cases of consumer misuse. But whatever our decision, we should be consistent about it.

Reply:
Theoretically yes. But the likelihood of a person specifying what a fake returns and using it as if it's something else is much lower than someone misusing the default "untyped" fake. 

It is very unlikely someone makes a fake which returns a number and assigns it to a string. 

It is likely someone makes a default fake and assigns it to non void.

It is difficult to measure the risk which is why I haven't written this as a blocking review. As it is indeed meant to be "void". It is just a question of if we want to risk the breaking change or not (that's what it is). 

Fakes have a much smaller usage than things like spies so I wouldn't worry over this being merged as is. But it is something to keep a very close eye on (if issues come up)

Reply:
OK, I guess the point of contention is around the scenaro that someone makes a default fake and assigns it to non void.

In my mind, if that is the case, then any time that fake is used in the implementation code, it will return the wrong thing -- if somebody uses `sinon.fake()` to make a function with signature `() => number`, it is never going to return a number, no matter what the types say, and therefore when their implementation calls it and expects a number, things will go wonky. I can't see how somebody could use it like this and _not_ experience a test failure.